src/lib/RoutingManager.ts

```diff
@@ -102,55 +136,62 @@ export default class RoutingManager {
     }
   }
 
-  dispose() {
-    if (this.renderURLFromState) {
-      this.instantSearchInstance.helper.removeListener(
-        'change',
-        this.renderURLFromState
-      );
-    }
-    this.router.dispose();
+  public init({ state }: { state: SearchParameters }): void {
```

Comment:
For some reasons the lifecycles hooks are not picked up by TypeScript even with:

```ts
... implement Widget {
 // ...
}
```

This is why these methods are typed.

Reply:
https://github.com/Microsoft/TypeScript/issues/1373